kad-dht/README.md

```diff
@@ -236,33 +257,119 @@ This process is repeated as many times per run as configuration parameter
 
 ## RPC messages
 
-_WIP (jheisey): consider just dumping a nicely formatted and simplified
-protobuf._
-
 See [protobuf
 definition](https://github.com/libp2p/go-libp2p-kad-dht/blob/master/pb/dht.proto)
 
 On any error, the entire stream is reset. This is probably not the behavior we
 want.
 
-* `FIND_NODE(key bytes) -> (nodes PeerInfo[])` Finds the `ncp` (default:6) nodes
-closest to `key` from the routing table and returns an array of `PeerInfo`s. If
-a node with id equal to `key` is found, returns only the `PeerInfo` for that
-node.
-* `GET_VALUE(key bytes) -> (record Record, closerPeers PeerInfo[])` If `key` is
-a public key (begins with `/pk/`) and the key is known, returns a `Record`
-containing that key. Otherwise, returns the `Record` for the given key (if in
-the datastore) and an array of `PeerInfo`s for closer peers.
-* `PUT_VALUE(key bytes, value Record) -> ()` Validates `value` and, if it is
-valid, stores it in the datastore.
-* `GET_PROVIDERS(key bytes) -> (providerPeers PeerInfo[], closerPeers
-PeerInfo[])` Verifies `key` is a valid CID. Returns `providerPeers` if in the
-providers cache, and an array of closer peers.
-* `ADD_PROVIDER(key, providerPeers PeerInfo[]) -> ()` Verifies `key` is a valid
-CID. For each provider `PeerInfo` that matches the sender's id and contains one
-or more multiaddrs, that provider info is added to the peerbook and the peer is
-added as a provider for the CID.
-* `PING() -> ()` Tests connectivity to destination node. Currently never sent.
+All RPC messages conform to the following protobuf:
+```protobuf
+// Record represents a dht record that contains a value
+// for a key value pair
+message Record {
+	// The key that references this record
+	bytes key = 1;
+
+	// The actual value this record is storing
+	bytes value = 2;
+
+	// Note: These fields were removed from the Record message
+	// hash of the authors public key
```

Comment:
At the network/RPC level, none of these last 3 fields exist in the Record, should they be removed from a spec?

Reply:
Yes they should be removed

Reply:
I am keeping `author` and `signature` in the protobuf definition, commented out, to make sure protobuf field numbers are not reused for a different purpose in the future.